hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/IncrementalRelation.scala

```diff
@@ -90,24 +78,24 @@ class IncrementalRelation(val sqlContext: SQLContext,
     throw new HoodieException("Incremental queries are not supported when meta fields are disabled")
   }
 
+  private val queryContext: IncrementalQueryAnalyzer.QueryContext =
+    IncrementalQueryAnalyzer.builder()
+      .metaClient(metaClient)
+      .startTime(optParams(DataSourceReadOptions.BEGIN_INSTANTTIME.key))
+      .endTime(optParams.getOrElse(DataSourceReadOptions.END_INSTANTTIME.key, null))
+      .rangeType(InstantRange.RangeType.OPEN_CLOSED)
+      .build()
+      .analyze()
+
+  private val commitsToReturn = List.concat(
+    queryContext.getArchivedInstants.asScala,
+    queryContext.getActiveInstants.asScala)
```

Comment:
The `instants` is already a concatenation of archived and active instants.

Reply:
`queryContext.instants` contains only the instant times.  There is one place below that needs to get the replaced instants from `commitsToReturn` based on the action type, so `queryContext.getArchivedInstants` and `queryContext.getActiveInstants` are used.  Another way is to make `queryContext.instants` store `HoodieInstant` instead of `String` of instant times.

Reply:
This is fixed now by adding a new API in `QueryContext` to get a list of `HoodieInstant`.